dspace-api/src/main/java/org/dspace/discovery/configuration/DiscoveryConfigurationService.java

```diff
@@ -69,7 +70,15 @@ public DiscoveryConfiguration getDiscoveryConfiguration(final String name) {
 
     public DiscoveryConfiguration getDiscoveryConfigurationByNameOrDso(final String configurationName,
                                                                        final IndexableObject dso) {
-        if (StringUtils.isNotBlank(configurationName) && getMap().containsKey(configurationName)) {
+
+        if (dso instanceof Community && configurationName.contentEquals("showUnDiscoverableItems")) {
```

Comment:
The normal procedure for defining separate configurations per community or collection is based on the discovery.xml map: https://github.com/DSpace/DSpace/blob/master/dspace/config/spring/api/discovery.xml#L41

Reply:
Yes, I knew this was not the right solution.  I just needed something to address the requirement pending feedback. I see how this is handled in discovery.xml. Will update.

Reply:
The requirement in the Jira issues reads:

> Review the scope/configurationName precedence, a search scoped to a community should look for a configuration named as the configurationName prefix + "." + handle failing back to the configurationName if missing

Having looked at this more, I'm not convinced that this requirement is necessary.  

But, assuming that it is needed, it can only be accomplished by looking for a configuration name in discovery.xml that is a composite of the current configurationName and community handle (from the query scope). That entails something like what was provisionally added here.

Again, it's not clear to me what purpose this serves, and without a scenario to work from it's probably best to ignore this requirement until it's clear why we need it.

Reply:
Agreed